components/messaging/CurrentConversation.tsx

```diff
@@ -3,25 +3,11 @@
 import { MessageType } from '@/utils/messaging/messagingTypes';
 import MessageCard from './MessageCard';
 import MessageForm from './MessageForm';
-import { useContext, useEffect } from 'react';
+import { useContext } from 'react';
 import useConversation from '../../app/(dashboard)/conversations/useConversation';
 
 const CurrentConversation: React.FC = () => {
-  const { allConversations, currentConversation, setCurrentConversation } =
-    useContext(useConversation);
-
-  useEffect(() => {
-    const highestId = Math.max(
-      ...allConversations.map((conversation) => conversation.conversation_id)
-    );
-
-    setCurrentConversation &&
-      setCurrentConversation(
-        allConversations?.filter(
-          (conversation) => conversation.conversation_id === highestId
-        )[0]
-      );
-  }, []);
```

Comment:
This is not a redundancy from updateCurrentConvo. The updateCurrentConvo function is a handler passed to each conversation card to update the currentConversation state when it is clicked. 

The useEffect in the `currentConversation` component is setting the current conversation _on render_. Without that useEffect we are rendering no conversation at all.

We can change to that behaviour but what we planned for originally was to always open one conversation on render which is why that useEffect was added in the first place.

Reply:
Okay this makes sense. So the code (as pictured above) was causing problems for us but I have since merged your refactor and I think we can have both (the currentConversation update on render and on conversation preview click)

The happy version:

```
useEffect(() => {
    setCurrentConversation && setCurrentConversation(allConversations[0]);
  }, [allConversations, setCurrentConversation]);

  useEffect(() => {
    const fetchMessagesForCurrentConversation = async () => {
      try {
        const { data: fetchedMessages } = await supabase
          .from('messages')
          .select('*')
          .eq('conversation_id', currentConversation?.conversation_id);

        setCurrentMessages(fetchedMessages ?? []);
      } catch (error) {
        console.error(`Failed to fetch messages from database: ${error}`);
        throw error;
      }
    };

    fetchMessagesForCurrentConversation();
  }, [currentConversation, setCurrentMessages]);
  
  ```